vivisect/qt/funcgraph.py

```diff
@@ -870,24 +885,26 @@ def _renderCodeBlock(self, data):
         '''
         if len(self.nodes):
             # render codeblock
-            node = self.nodes.pop(0)
-            cbva = node[1].get('cbva')
-            cbsize = node[1].get('cbsize')
-            self.mem_canvas.renderMemory(cbva, cbsize, self._renderCodeBlock)
-
-            # update _xref_cache_bg (made "real" when rendering complete)
-            endva = cbva + cbsize
-            while cbva < endva:
-                for xref in self.vw.getXrefsFrom(cbva):
-                    xrfr, xrto, xrtype, xrflag = xref
-                    self._xref_cache_bg.add(xrto)
-                    #logger.debug("adding 0x%x -> 0x%x to _xref_cache", xrfr, xrto)
-
-                lva, lsz, ltp, ltinfo = self.vw.getLocation(cbva)
-                cbva += lsz
+            blocks = []
+            for node in self.nodes:
+                cbva = node[1].get('cbva')
+                cbsize = node[1].get('cbsize')
+                blocks.append([cbva, cbsize])  # yes this has to be a list so javascript plays nicely
+                #self.mem_canvas.renderMemory(cbva, cbsize, self._renderCodeBlock)
+
+                # update _xref_cache_bg (made "real" when rendering complete)
+                endva = cbva + cbsize
+                while cbva < endva:
+                    for xref in self.vw.getXrefsFrom(cbva):
+                        xrfr, xrto, xrtype, xrflag = xref
+                        self._xref_cache_bg.add(xrto)
+
+                    lva, lsz, ltp, ltinfo = self.vw.getLocation(cbva)
+                    cbva += lsz
+            self.mem_canvas.renderMemory(blocks, self._getNodeSizes)
```

Comment:
callback `_getNodeSizes` instead of `_renderCodeBlock`?
i don't follow, sorry.  can you explain?

Reply:
well done.  i look forward to merging this in.